lib/RealWlanInterface.cpp

```diff
@@ -384,6 +384,7 @@ void RealWlanInterface::OnScanComplete()
         if (m_scanPromise)
         {
             m_scanPromise->set_value({std::move(results), ScanStatus::Completed});
```

Comment:
Exactly, it sets the shared state that will let any future linked to this promise to complete. In this case, it will unblock the scan request in the operation handler.